### PR TITLE
T21717

### PR DESCRIPTION
**Pull request by tchx84**
https://phabricator.endlessm.com/T21717

### CODE REVIEW

File: js/app/modules/banner/dynamic.js [hunk 1]
```diff
@@ -44,6 +44,13 @@ var Dynamic = new Module.Class({
          */
         'mode': GObject.ParamSpec.string('mode', 'mode', 'mode',
             GObject.ParamFlags.READWRITE | GObject.ParamFlags.CONSTRUCT, 'text'),
+         /**
+         * Property: Layout
+         *
+         * The layout of the logo in full mode, it can be auto, horizontal or vertical.
+         */
+        'layout': GObject.ParamSpec.string('layout', 'layout', 'layout',
```
Comment (ptomato) [inline js/app/modules/banner/dynamic.js]: Drive-by comment: it would be good to have an enum here; for the YAML file it makes no difference.

Comment (tchx84) [reply js/app/modules/banner/dynamic.js]: As we spoke, we don't do this anywhere and seems it can't be done (without creating a C enum I supposed).

File: js/app/widgets/dynamicLogo.js [hunk 2]
```diff
@@ -139,6 +147,17 @@ var DynamicLogo = new Knowledge.Class({
         return this._mode;
     },
 
+    set layout(value) {
+        if (this._layout === value)
+            return;
+        this._layout = value;
+        this.queue_draw();
```
Comment (xjuan) [inline js/app/widgets/dynamicLogo.js]: This should actually call queue_resize() instead of draw

Comment (tchx84) [reply js/app/widgets/dynamicLogo.js]: This is a weird widget, it receives an imposed size and it will always make sure that everything fits inside the size. Therefore, changing the layout won't need a resize, because the widget will simply draw in whatever space has available.

**Comment by xjuan** [comment]
CR Pass
